hack/update-pkger.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source $(dirname $0)/../vendor/knative.dev/hack/library.sh
+
+# Hack: Remove the build tag before running pkger
+tools="${REPO_ROOT_DIR}/hack/tools.go"
+first="$(head -n1 "$tools")"
+sed -i '1d' "$tools"
+
+go run ./vendor/github.com/markbates/pkger/cmd/pkger
+
+# Hack: Restore build tag.
+echo -e "$first\n$(cat "$tools")" > "$tools"
```

Comment:
The `sed` command fails for developers with the default BSD `sed` on OSX with the somewhat cryptic error `extra characters at the end of d command`.  Perhaps we could use something which works on both platforms by default?
```suggestion
# Hack: touch a non-tagged file so that pkger doesn't complain
echo -e "package tools" > "$REPO_ROOT_DIR/hack/package.go"

go run ./vendor/github.com/markbates/pkger/cmd/pkger

# Hack: remove touched file.
rm "$REPO_ROOT_DIR/hack/package.go"
```

Reply:
Alternately, we could just add the `package.go` file to the hack directory with an explanatory comment such as `// Required for pkger` or some such, and avoid the programmatic creation/removal above.  :shrug:

Reply:
`gnu-sed` is required for Knative scripts anyway FWIW.

Sadly we can't just not have `tools.go` not have a build-constraint, because that then breaks the license checking tool :joy:

Reply:
We're not running the knative-X action scripts locally during development.  This proposed change does not, to my understanding, have any effect on tools.go nor the license checking tool because `tools.go` remains untouched.  Please enlighten me.

Reply:
The point with `sed`, `bash` and MacOSX is: you'll need gnu-sed and bash4+ for the Knative scripts anyway. Sure, you might somewhat be able to get around it by only using the `make` tooling, but as soon as you call `update-deps` (and you will have to do that locally), you'll need these anyway.